generic_permissions/visibilities.py

```diff
@@ -60,6 +60,33 @@ def get_attribute(self, instance):
         return queryset
 
 
+class VisibilitySerializerMixin:
+    def validate(self, *args, **kwargs):
+        validated_data = super().validate(*args, **kwargs)
+
+        if not self.instance:
+            return validated_data
+
+        # Incoming patches can have a subset of all relations, so we need to
+        # keep the existing relations which are not included in the request.
+        for key, field in self.fields.items():
+            if (
+                type(field) is not VisibilityManyRelatedField
+                or key not in validated_data
+            ):
+                continue
+
+            # Find the relations which the request can include.
+            queryset = getattr(self.instance, key).all()
+            for handler in VisibilitiesConfig.get_handlers(queryset.model):
+                queryset = handler(queryset, self.context)
+
+            # Add remaining relations which can not be included in the request.
+            validated_data[key] += getattr(self.instance, key).exclude(pk__in=queryset)
+
+        return validated_data
+
+
 class VisibilityRelatedFieldMixin:
```

Comment:
I would recommend adding a check here to ensure the serializer has the (strongly recommended) `VisibilitySerializerMixin` in it's base class list:

```suggestion
class VisibilityRelatedFieldMixin:
    def bind(self, field_name, parent):
        parent_cls = type(parent)
        if VisibilitySerializerMixin not in parent_cls.mro():
            raise RuntimeWarning(
                f"To avoid data loss, consider using VisibilitySerializerMixin in {parent_cls.__name__}")
        return super().bind(field_name, parent)

```

Reply:
Added